torch/csrc/generic/Storage.cpp

```diff
@@ -104,7 +114,10 @@ static PyObject * THPStorage_(pynew)(PyTypeObject *type, PyObject *args, PyObjec
         self->cdata->unsafe_data<scalar_t>()[i] = value;
 #else
         // TODO: this might be slow - consider batched updates?
-        THCStorage_(set)(LIBRARY_STATE self->cdata, i, value);
+        storage_set(
+          at::unsafeStorageFromTH(self->cdata, /*retain=*/true),
```

Comment:
Do you need `unsafeStorageFromTH here, or is `cdata` StorageImpl already?

Reply:
This creates a `at::Storage` from `c10::StorageImpl*`.